### PR TITLE
Fixes #31880 - correctly use SSL for qpid connections

### PR DESCRIPTION
**Pull request by jturel**
Correctly sets up the SSL connection and handles closing qpid connections a little better. I need to give this a more thorough test but my dev env is in an odd state..

To be tested in conjunction with https://github.com/theforeman/puppet-katello/pull/384 and https://github.com/theforeman/puppet-foreman_proxy_content/pull/327/files

### CODE REVIEW

File: lib/katello/engine.rb [hunk 1]
```diff
@@ -58,8 +58,12 @@ class Engine < ::Rails::Engine
...
         :agent => {
           :client_queue_format => 'pulp.agent.%s',
-          :broker_url => 'amqp:ssl:localhost:5671',
-          :event_queue_name => 'katello.agent'
+          :event_queue_name => 'katello.agent',
+          :broker_url => 'amqps://localhost:5671',
+          # puppetize these
+          :broker_ssl_cert_file => '/etc/pki/katello/qpid_router_client.crt',
+          :broker_ssl_key_file => '/etc/pki/katello/qpid_router_client.key',
+          :broker_ssl_ca_file => '/etc/pki/katello/certs/katello-default-ca.crt'
```
Comment (ehelms) [inline lib/katello/engine.rb:66]: We do not need to introduce special certificates here for this. We can use the main Foreman client certificates which is a move we are making in general:

https://github.com/theforeman/puppet-foreman/blob/master/templates/settings.yaml.erb#L24-L26

File: app/lib/katello/qpid/connection.rb [hunk 2]
```diff
@@ -49,17 +49,18 @@ def on_message(_delivery, message)
       end
 
       class Receiver < ::Qpid::Proton::MessagingHandler
-        def initialize(url, address, handler)
+        def initialize(url, connection_options, address, handler)
```
Comment (ehelms) [inline app/lib/katello/qpid/connection.rb:52]: The Sender method signature also needs an update -- https://github.com/Katello/katello/pull/9166/files#diff-d5d0f092b1171d0232fd04dc54b5b9f9dc095735857a4f8e7aecf0bd1f35dabbR7

**Review by ehelms — APPROVED** [review]
From a testing it on a nightly perspective, this looked good to me. I opened https://github.com/theforeman/puppet-katello/pull/385 to correspond with the needed URL change.

**Comment by jturel** [comment]
[test katello]

**Comment by jturel** [comment]
[test katello]

**Pull request merged**